### PR TITLE
GH Action for push and pull requests

### PR DESCRIPTION
**Pull request by imbrn**
Add Github Actions push and pull request jobs for testing the application

## Type of change

- [x] Other (please, feel free to specify what kind of change it is)

## Checklist:

- [x] I have created my branch from a recent version of `master`
- [x] The code is clean and easy to understand

### CODE REVIEW

**Comment by barfurth** [comment]
If we want to use `npm ci` we need to publish the lock file. This is what [yarn suggests anyways](https://classic.yarnpkg.com/blog/2016/11/24/lockfiles-for-all/). This will also be a problem with the publish action I think?

**Comment by imbrn** [comment]
> If we want to use `npm ci` we need to publish the lock file. This is what [yarn suggests anyways](https://classic.yarnpkg.com/blog/2016/11/24/lockfiles-for-all/). This will also be a problem with the publish action.

Oh, absolutely. I'm still working on that.

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```
Comment (barfurth) [inline .github/workflows/push.yml:19]: Not sure why, but this is a different format than in the publish action. There it's only the major version - here it's `.x` for some reason?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
```
Comment (barfurth) [inline .github/workflows/push.yml:23]: This one also has a title where the publish action does not. They are styled quite differently. They both work, it's simply a matter of consistency.

Comment (imbrn) [reply .github/workflows/push.yml:23]: Great point! Actually, I have to edit the publish action to use `yarn` instead. I'll make it consistent as well in the next PR. Thank you.

**Pull request merged**